csvuploader/src/main/java/com/tweek/service/FileSystemStorageService.java

```diff
@@ -50,6 +51,23 @@ public String saveFile(MultipartFile file) {
 		}
 	}
 
+	private String transformFilename(String identifier, String filename) {
+
+		filename = filename.trim().toLowerCase().replaceAll("\\s+", "_");
+
+		if(filename.contains(".")) {
+			filename = filename.substring(0, filename.lastIndexOf("."));
+		}
+
+		return filename + "_" + identifier;
+	}
+
+	public void saveFile(String identifier, MultipartFile file) throws IOException {
```

Comment:
I've already modified this method in the latest commit on master. So now it accepts write key and collectionName. Would you please pull the latest from master and check?